content/advent-2016/contributing-to-the-go-project.md

```diff
@@ -0,0 +1,218 @@
+```
+
+This will open a commit message file in your editor (using `$EDITOR`).
+
+There are some conventions that should be followed for commit messages,
+including:
+
+- commit message should be prefixed with package name
+- a one-line summary of the change
+- if needed, a detailed description of the change (written in complete
+sentences with proper punctuation)
+- the phrase "Fixes ###", where ### is the Go issue you are resolving
+
+An example of a commit message in this style, from the Contribution
+Guidelines:
+
+```
+math: improve Sin, Cos and Tan precision for very large arguments
+
+The existing implementation has poor numerical properties for
+large arguments, so use the McGillicutty algorithm to improve
+accuracy above 1e10.
+
+The algorithm is described at http://wikipedia.org/wiki/McGillicutty_Algorithm
+
+Fixes #159
+```
+
+If you wish to make further changes, use normal `git` commands and
+just run `git change` again to amend your commit.
+
+Finally, submit your change to Gerrit by running:
+
+```
+$ git mail
+```
+
+The output of `git mail` will print a link to where your change can be
+found, such as:
+
+```
+remote: New Changes:
+remote:   https://go-review.googlesource.com/99999 math: improved Sin, Cos and Tan precision for very large arguments
+```
+
+## Gerrit code review
+
+Now that your change has been submitted, it can be reviewed by a member
+of the Go team and others in the community.  Code review comments may
+be addressed by amending your changes using the process above.
+
+At this point, any number of actions can take place with your change.
+Typically, a member of the Go team will comment with a `Code-Review`
+label, which can be interpreted as follows:
+
+- `-2`: I am strongly against this change and will almost certainly not
+be persuaded otherwise
+- `-1`: I disagree with this change, but could be persuaded otherwise
+- `+1`: this change looks good to me, but someone else must approve
+- `+2`: this change looks good to me, and is ready for submission
+
+Depending on the `Code-Review` label applied and the ensuing discussion,
+your change may or may not be accepted into the project.
+
+In particular, if you submit a change with no issue filed and no
+discussion beforehand, you can be almost certain it will be rejected.
+
+Don't be disheartened!  It can take some practice to follow the process,
+but once you've worked through it a couple of times, it's a breeze!
+
+## Summary
+
+In summary, contributing to the Go project can be intimidating, but is
+an excellent way to give back to the Go community and to gain experience
+working on a large open source project, used by thousands of people all
+over the world.
+
+Again, don't be disheartened if your first change is rejected.  Be polite
+and ask for clarification if needed.  If you disagree with a comment,
+state your case concisely and make sure that no misunderstandings take
+place.
+
+With practice and perseverance, you too can make a difference!
```

Comment:
I would rework or remove this sentence; this sentence currently has an arrogant feel to it.